File: sprint4/By languages/JS/lesson1code4.js
class Map {
  constructor() {
    this.pairs = [];
  }

  get(key) {
    for (let pair of this.pairs) {
      if (pair.key === key) { // пара с указанным ключом найдена
        return pair.value;
      }
    }
    return null;
  }

  set(key, value) {
    for (let pair of this.pairs) {
      if (pair.key === key) { // пара с указанным ключом найдена
        // обновить значение в найденной паре
        pair.value = value;
        return;
      }
    }
    // пара с заданным ключом не найдена
    // добавить пару (key, value) в pairs
    this.pairs.push({key: key, value: value});
  }
}

File: sprint4/By languages/swift/lesson1code4.swift
class Pair {
    let key: String
    var value: Any
    
    init(key: String, value: Any) {
        self.key = key
        self.value = value
    }
}

class Map {
    private var pairs: [Pair] = []
    
    func get(key: String) -> Any? {
        for pair in pairs {
            if pair.key == key {
                return pair.value
            }
        }
        return nil
    }
    
    func set(key: String, value: Any) {
        for pair in pairs {
            if pair.key == key {
                pair.value = value
                return
            }
        }
        let newPair = Pair(key: key, value: value)
        pairs.append(newPair)
    }
}